pinax/ratings/tests/test_views.py

```diff
@@ -0,0 +1,98 @@
+from django.contrib.contenttypes.models import ContentType
+
+from .models import Car
+from .test import TestCase
+
+
+class RateViewTest(TestCase):
+    # tests for the RateView
+
+    def setUp(self):
+        self.test_user = self.make_user(username="test_user")
+        self.forester = Car.objects.create(name="Subaru Forester")
+
+    def test_post_rating_with_invalid_content_type_id(self):
+        # test line with code snippet:
+        #   get_object_or_404(ContentType, pk=self.kwargs.get("content_type_id"))
+        #
```

Comment:
Yes, it's good to have a comment explaining the test goal!

Two comments here: Please use the "docstring" triple double-quote format as described in the Pinax [How to Contribute -
 Coding Style](http://pinaxproject.com/pinax/how_to_contribute/):

```
"""
docstring
"""
```

Secondly, it's not usually good practice to show lines from the tested code. Tests really shouldn't know what is going on in the underlying tested code, but rather the expected result. In this case, the test is proving the view returns 404 if the generic object is not found. There's no need to copy the view code into the docstring. `"""Ensure view returns 404 if object is not found"""` is a reasonably good docstring here.

This commentary applies to the other test methods as well. Their initial comments should be converted to docstrings.

Docstrings are useful because often test runners display the failing test docstring, giving you a better idea about what failed than just the test name.

Docstrings are also helpful to developers trying to understand the package as a whole.

Reply:
@grahamu I have converted the comments to docstrings as per pinax coding style guidelines. I have also removed the comment that was showing the code under test.